cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/sourcecontrol/SourceControlHelper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.sourcecontrol;
+
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.sourcecontrol.SourceControlException;
+import io.cdap.cdap.sourcecontrol.operationrunner.PullAppResponse;
+import java.util.Collection;
+
+/**
+ * Provides various helper methods for source control operations. Would be implemented for both
+ * running in app-fabric and running in workers.
+ */
+public interface SourceControlHelper {
```

Comment:
this isn't really source control specific. It's more like an ApplicationManager or something like that.

Reply:
It may not only include application specific methods e.g. `getRepositoryConfig`. The idea is to abstract every interaction with db/app-fabric